### PR TITLE
Feature/ch145872/h3 library to return hexadecimal

### LINKED ISSUE #54: ST_BOUNDARY generates error in BigQuery
Hi. I am using the bqcartoeu UDFs and have a table with H3 cell values generated from point data. I want to visualize different statistics on these cells, for example, number of observations per cell. I am trying to generate the geometry for the cells using:

`select bqcartoeu.h3.ST_BOUNDARY(bqcartoeu.h3.H3_FROMHEX(col_with_h3_hex_string))`

This gives an error:

`ST_GeogFromText failed: Invalid polygon loop: Edge 5 crosses edge 7 `

Unfortunately, I cannot localize which H3 values are causing the error, but I want to ask if the SAFE prefix could be added to the UDF where you are calling ST_GeogFromText.

I guess I am thinking it would make sense here?

https://github.com/CartoDB/carto-spatial-extension/blob/afb66d77dde3f1773a9bb877bd793cfcac30ae46/h3/bq/sql/ST_BOUNDARY.sql#L32

Thanks!

### PR DESCRIPTION
**Pull request by vdelacruzb**
This also Closes #54

### CODE REVIEW

**Comment by vdelacruzb** [comment]
CI is failing, so I will apply some changes in placekey as well.

File: h3/bq/sql/KRING.sql [hunk 1]
```diff
@@ -4,25 +4,18 @@
...
     LANGUAGE js
     OPTIONS (library=["@@H3_BQ_LIBRARY@@"])
 AS
 """
-    if (index_lower == null || index_upper == null || distance == null || distance < 0)
+    if (!index || distance == null || distance < 0)
         return null;
-    const h3IndexInput = [Number(index_lower), Number(index_upper)];
-    if (!h3.h3IsValid(h3IndexInput))
+
```
Comment (Jesus89) [inline h3/bq/sql/KRING.sql:16]: Extra space

File: h3/bq/sql/ST_BOUNDARY.sql [hunk 2]
```diff
@@ -4,19 +4,20 @@
...
     LANGUAGE js
     OPTIONS (library=["@@H3_BQ_LIBRARY@@"])
 AS
 """
-    if (index_lower == null || index_upper == null)
+    if (!index)
         return null;
-    const h3IndexInput = [Number(index_lower), Number(index_upper)];
-    if (!h3.h3IsValid(h3IndexInput))
+
```
Comment (Jesus89) [inline h3/bq/sql/ST_BOUNDARY.sql:16]: Extra space

File: h3/bq/sql/TOCHILDREN.sql [hunk 3]
```diff
@@ -4,25 +4,18 @@
...
     LANGUAGE js
     OPTIONS (library=["@@H3_BQ_LIBRARY@@"])
 AS
 """
-    if (index_lower == null || index_upper == null)
+    if (!index)
         return null;
-    const h3IndexInput = [Number(index_lower), Number(index_upper)];
-    if (!h3.h3IsValid(h3IndexInput))
+
```
Comment (Jesus89) [inline h3/bq/sql/TOCHILDREN.sql:16]: Extra space

File: h3/bq/doc/REFERENCE.md [hunk 4]
```diff
@@ -17,17 +17,17 @@ Returns the H3 cell index that the point belongs to in the required `resolution`
...
 
 ```sql
 SELECT bqcarto.h3.ST_ASH3(ST_GEOGPOINT(40.4168, -3.7038), 4);
--- 596645165859340287
+-- 847b59dffffffff
 ```
 
 {{% bannerNote type="note" title="tip"%}}
-If you want the HEX representation of the cell, you can call [H3_ASHEX](#h3_ashex) with the output integer.
+If you want the HEX representation of the cell, you can call [H3_FROMINT](#h3_ashex) with the output integer.
```
Comment (Jesus89) [inline h3/bq/doc/REFERENCE.md:30]: I would change this to:

```
If you want the INT representation of the cell, you can call [H3_ASINT](#h3_asint) with the output integer.
```

**Review by Jesus89 — APPROVED** [review]
Good job! The code looks cleaner now following the standard

There are minor changes, and I would use the same notation for H3 indexes ("837b5dfffffffff")

I see this PR includes the ST_BOUNDARY fix, so I would merge this one first: https://github.com/CartoDB/carto-spatial-extension/pull/55

**Pull request merged**